Mathlib/AlgebraicGeometry/EllipticCurve/Jacobian.lean

```diff
@@ -1503,6 +1561,60 @@ end Point
 
 end Affine
 
+section Map
+
+lemma comp_fin3 {S} (f : R → S) (X Y Z : R) : f ∘ ![X, Y, Z] = ![f X, f Y, f Z] := by
+  ext i; fin_cases i <;> rfl
```

Comment:
Can you either
* remove this lemma in favour of lemmas about `vecCons` and `vecEmpty`
* make this lemma `private`

Reply:
It's primarily used in a different file (`ZSMul.lean`) so I don't think it can be made private ...? We have four similar convenience lemmas starting from [WeierstrassCurve.Jacobian.fin3_def](https://leanprover-community.github.io/mathlib4_docs/Mathlib/AlgebraicGeometry/EllipticCurve/Jacobian.html#WeierstrassCurve.Jacobian.fin3_def) and I don't think they're outrageous ...

Reply:
Yeah, I think this is sensible just for our purposes, as long we hide them in the `Jacobian` namespace.

Reply:
I would move this to right below `fin3_def` or `smul_fin3`, but it's not a huge deal. I will eventually split this file into smaller files and relocate a lot of things, so I might as well do that then.

Reply:
Why is my first option not viable?

Reply:
What are we supposed to do with the first option? It does seem that `comp_vecCons` and `comp_vecEmpty` are missing, but IMO that should be another PR, because I do want `comp_fin3` for precise rewrite using a single lemma, and its proofs can't be made shorter with the `comp_vecCons/Empty` lemmas.

(Lemmas like [WeierstrassCurve.Jacobian.smul_fin3](https://leanprover-community.github.io/mathlib4_docs/Mathlib/AlgebraicGeometry/EllipticCurve/Jacobian.html#WeierstrassCurve.Jacobian.smul_fin3) can't be generalized to arbitrary vectors since different coordinate has different weighting (exponent).)

Reply:
This lemma already exists in the reflection library